app/controllers/modules/course/announcements_module.rb

```diff
@@ -5,7 +5,10 @@ class Course::AnnouncementsModule
     [
       {
         title: I18n.t('course.announcements.sidebar_title'),
-        unread: 0
+        unread:
+          if @course && @current_user
+            @course.announcements.unread_by(@current_user).count
```

Comment:
To me I think a general function is needed here, because for all the modules in course their unread count will in the same form of <code>@course.xxx.unread_by(@current_user).count</code>

And this function should return 0 when <code>@course || @current_user </code> is not present.

@lowjoel Thoughts ?